### DISCUSSION TITLE
Path for embedding the image not exist when deployed to Vercel

### DISCUSSION THREAD

**Discussion by Wavesolid** [eleith/emailjs#313] [questions] [answered]
I want to send an Image from `public` folder, it works on my local server, but after I deployed it to Vercel, the server can't find the path folder and return me an error that the image is not exist. Here my code:
![image](https://user-images.githubusercontent.com/58335667/163706142-b5a4d830-9f60-41a4-88ce-c5979b44f559.png)

I already tried with different approach like:
`public/qrcodes/sadam.png`
`/qrcodes/sadam.png`
`${process.cwd()}/public/qrcodes/sadam.png`
it's still not works. Does this issues are coming from this dependencies or Vercel itself? thank you in advance.

**Comment by zackschuster** [comment, marked as answer]
this looks like an issue with the file url. does something like `new URL('./public/qrcodes/sadam.png', import.meta.url)` work for you?

**Reply by Wavesolid** [reply]
@zackschuster no it's not working instead gave me a new error when `build` the app. Error message is `Module not found: Can't resolve './public/qrcodes/sadam.png' in '/vercel/path0/pages/api'`. I also tried with `path: new URL('/qrcodes/sadam.png', process.env.BASE_URL)`, which `process.env.BASE_URL` contain my base url and this `new URL()` has value like this `https://weddingneysagintano.vercel.app/qrcodes/sadam.png` that you can see the picture is exist.

**Reply by zackschuster** [reply]
what does `import.meta.url` resolve to? i would think it would point to `'/vercel/path0'` or something similar. basically it seems like the assets in `/public` are being stored in a place different from where you're expecting it. i've not used vercel myself but [they do talk about something called asset hoisting](https://vercel.com/guides/using-express-with-vercel) which might be where your problems lie?

**Reply by Wavesolid** [reply]
after doing several research, I just found myself that Next.Js in Vercel is not supporting readfile for any serveless function, which always resulting `file is not exist`.

**Reply by zackschuster** [reply]
ah, it's a serverless function. it looks like [vercel can package flat files for you](https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions). hopefully that works? at any rate, it's definitely not a problem with the library, so i'll close this down. thanks for reaching out!
